src/service_discovery_setup.js

```diff
@@ -0,0 +1,274 @@
+const dnssd = require('dnssd');
+const config = require('config');
+const Settings = require('./models/settings');
+
+// Internal dependancies
+
+const server = new DNSserviceDiscovery();
+
+
+/**
+ * Our DNS Service Discovery object.
+ *
+ * Creates an object which contains the 'profile' used to have a mDNS service
+ * discovery process running. The object will be instanciated with default
+ * values to ensure it will always work no matter DB or Config state.
+ *
+ * It provides methods to interact with the object. This object is created
+ * quite early on since the starting process in app.js and parts of the UI in
+ * settings_controller.js need to use its methods.
+ *
+ *
+ * @param {}
+ */
+
+function DNSserviceDiscovery() {
+  this.serviceState = false;
+  this.port = config.get('ports.https');
+  this.localDomain = config.get(
+    'settings.defaults.domain.localControl.mdnsServiceDomain');
+  this.localName= config.get(
+    'settings.defaults.domain.localControl.mdnsServiceName');
+  this.protocol= config.get(
+    'settings.defaults.domain.localControl.mdnsTxt.protocol');
+  this.power= config.get(
+    'settings.defaults.domain.localControl.mdnsTxt.power');
+  this.description= config.get(
+    'settings.defaults.domain.localControl.mdnsTxt.desc');
+  const txt = {desc: this.description,
+               protocols: this.protocol,
+               power: this.power};
+  const options = {name: this.localName, host: this.localDomain, txt};
+
+  // Initialize our object and make sure it's not started on object creations.
+  this.dnssdHandle = dnssd.Advertisement(dnssd.tcp('http'), this.port, options);
+  this.dnssdHandle.stop();
+}
+
+
+/**
+ * A small helper method.
+ *
+ * Used to update the internal state of the mDNS service discovery.
+ * Then restart it depending on the boolean 'state' parameter.
+ *
+ * @param {boolean}: True starts the broadcast service.
+ *                   False stops the broadcast service
+ */
+
+DNSserviceDiscovery.prototype.startService = function(state) {
+  this.serviceState = !!state;    // Make sure it's a boolean value
+  if (state) {
+    this.dnssdHandle.start();
+  } else {
+    this.dnssdHandle.stop();
+  }
+  console.log(`Service Discover state changed to: ${state}`);
+};
+
+
+/**
+ * A small helper method.
+ *
+ * Used to update the local name of the mDNS service discovery. It does this
+ * by updating it's localDomain name, stopping the old service and then
+ * creating a new dnssd object.
+ *
+ * @param {local DNS name. e.g. Myhome-iot}
+ */
+
+DNSserviceDiscovery.prototype.setLocalDomain = function(localDomain) {
+  // TODO check the localDomain name is a legit mDNS name
+  // Check any letter or number a-Z,A-Z, 0-9 and '-' any number of times with
+  // a length less than 63 or contain www and api
+  localDomain = localDomain.toLowerCase();
+  const re = new RegExp(/^([a-z0-9]|[a-z0-9][a-z0-9-]*[a-z0-9])$/);
+  const valid = re.test(localDomain) && !['api', 'www'].includes(localDomain) &&
```

Comment:
You can remove this check: `!['api', 'www'].includes(localDomain)`

That was very specific to the tunnel service domain.

Reply:
done.